src/renderer/api/kube-api.ts

```diff
@@ -229,8 +229,11 @@ export class KubeApi<T extends KubeObject = any> {
       });
 
       if (this.apiVersionPreferred) {
-        Object.defineProperty(this, "apiBase", { value: this.getUrl() });
+        const apiBase = this.getUrl();
+
+        Object.defineProperty(this, "apiBase", { value: apiBase });
         apiManager.registerApi(this.apiBase, this);
+        apiManager.updateStoreKey(this.objectConstructor.apiBase);
```

Comment:
Maybe better to do all the job for updating the keys inside `registerApi` since we pass `this` already?
Something like: 

```ts
apiManager.registerApi(this.apiBase, this, {update: true} /*options={}*/);
```

Also, purpose of `updateStoreKey` is not clear for me (why / when should I use it).